src/core/reference/src/op/random_uniform.cpp

```diff
@@ -73,6 +41,21 @@ float uint32_to_float(uint32_t x) {
     return out_val.f - 1.0f;
 }
 
+float uint32_to_float_pytorch(uint32_t x) {
+    // float32 is formatted as follows: sign(1 bit) exponent(8 bits) mantissa(23 bits). The value is interpreted
+    // The value is interpreted using following formula:
+    // (-1)^sign * 1, mantissa * 2 ^ (exponent - 127)
+    // Here we set the following values:
+    // sign = 0
+    // exponent = 127, for obtaining a zero exponent.
+    // mantissa = 23 right bits from generated uint32 random value.
+    auto MASK = static_cast<uint32_t>((static_cast<uint64_t>(1) << std::numeric_limits<float>::digits) - 1);
```

Comment:
If I understand correctly, this behavior is a good practice in case some rare system has a float length higher than 27 bits. But I agree it's rather redundant, so I removed it